src/event.rs

```diff
@@ -906,15 +906,7 @@ where
 				};
 
 				match self.payment_store.update(&update) {
-					Ok(true) => (),
-					Ok(false) => {
-						log_error!(
-							self.logger,
-							"Payment with ID {} couldn't be found in store",
-							payment_id,
-						);
-						debug_assert!(false);
-					},
+					Ok(_) => (),
```

Comment:
Perhaps document why we can ignore this result and mention the replay possbility?

Reply:
I actually meant is to explain why this function is called multiple times and that that is the reason for ignoring the result value.

Reply:
Someone might still wonder why the return value is ignored, and to understand this they need to know that the method can be called multiple times. Perhaps it can work too as a comment on the method itself.

It is a question that I had while reviewing, so other might too.